app/uk/gov/hmrc/hec/services/TaxCheckService.scala

```diff
@@ -70,6 +80,16 @@ class TaxCheckServiceImpl @Inject() (
     taxCheckStore.store(taxCheck).map(_ => taxCheck)
   }
 
+  def updateTaxCheck(
```

Comment:
i will generalise it, so that we can use it in future if we want to update some other fields and keep it public. what do you think?

Reply:
we can always make it public when the need arises, until then I think it's better to keep it private

Reply:
@AndyHWChung  what do you think?

Reply:
is `updateTaxCheck` not being used elsewhere? If it's not then we probably don't need it exposed as a public thing in the trait - if we end up just updating one thing in the future we could just reuse the more general `updateAllHecTaxCheckStatus` (which should probably be maybe be called something more general like `updateAllHecTaxChecks` since it's not just updating sent status in particular)

Reply:
Yes, that's what i mentioned above, that i will generalise it, it can be used to update any other field and not specific IsExtracted field  and then we can keep it as public.

Reply:
ok, i have removed it, i think we can use the existing saveTaxCheck for updating as well. Just pass on the updated HECTaxCheckData. `updateAllHecTaxCheck` is now calling saveTaxCheck

Reply:
`updateHecTaxCheck` is now a general method - can be used to update any other values in `HECTaxCheck` and is not specific to isExtracted field

Reply:
[updated](https://github.com/hmrc/hec/pull/21/commits/bf9a98eb364aba07cad6963704fc9741732bd915)